src/bionics.cpp

```diff
@@ -546,11 +547,28 @@ bool player::activate_bionic( int b, bool eff_only )
         if( !has_cable ) {
             add_msg_if_player( m_info,
                                _( "You need a jumper cable connected to a vehicle to drain power from it." ) );
+        } else {
+            const optional_vpart_position vp_pos = g->m.veh_at( pos() );
+            if( vp_pos ) {
+                vehicle *const veh = &vp_pos->vehicle();
+                const cata::optional<vpart_reference> seat = vp_pos.part_with_feature( "BOARDABLE" );
+                if( this == veh->get_passenger( seat->part_index() ) ) {
```

Comment:
`seat` is of type `cata::optional<vpart_reference>`. The `optional` means it may not actually contain a `vpart_reference`, dereferencing it is like dereferencing a null pointer - it triggers UB. Always check that the optional contains a value before using it.

Reply:
Fixed, thanks.